--- a/extensions/ACCOUNT/xt_ACCOUNT.c
+++ b/extensions/ACCOUNT/xt_ACCOUNT.c
@@ -37,7 +37,7 @@
 
 #include <net/route.h>
 #include "xt_ACCOUNT.h"
-#include "compat_xtables.h"
+#include "../compat_xtables.h"
 
 #if (PAGE_SIZE < 4096)
 #error "ipt_ACCOUNT needs at least a PAGE_SIZE of 4096"
--- a/extensions/LUA/controller.h
+++ b/extensions/LUA/controller.h
@@ -19,11 +19,11 @@
 #ifndef CONTROLLER_H_
 #define CONTROLLER_H_
 
-#include "stdlib.h"     /* wrapper */
-#include "string.h"     /* wrapper */
-#include "lua.h"
-#include "lualib.h"
-#include "lauxlib.h"
+#include "lua/include/stdlib.h"     /* wrapper */
+#include "lua/include/string.h"     /* wrapper */
+#include "lua/lua.h"
+#include "lua/lualib.h"
+#include "lua/lauxlib.h"
 
 #if defined(__KERNEL__)
 #include <linux/skbuff.h>
--- a/extensions/LUA/lua/lapi.c
+++ b/extensions/LUA/lua/lapi.c
@@ -4,7 +4,7 @@
 ** See Copyright Notice in lua.h
 */
 
-#include <string.h>
+#include <linux/string.h>
 
 #define lapi_c
 #define LUA_CORE
--- a/extensions/LUA/lua/lauxlib.c
+++ b/extensions/LUA/lua/lauxlib.c
@@ -5,11 +5,11 @@
 */
 
 #if !defined(__KERNEL__)
-#include <ctype.h>
-#include <errno.h>
-#include <stdio.h>
-#include <stdlib.h>
-#include <string.h>
+#include "include/ctype.h"
+#include "include/errno.h"
+#include "include/stdio.h"
+#include "include/stdlib.h"
+#include "include/string.h"
 #else
 #include <linux/ctype.h>
 #include <linux/errno.h>
--- a/extensions/LUA/lua/lauxlib.h
+++ b/extensions/LUA/lua/lauxlib.h
@@ -9,11 +9,11 @@
 #define lauxlib_h
 
 
-#include <stddef.h>
+#include <linux/stddef.h>
 #include <linux/slab.h>	/* for kmalloc and kfree when allocating luaL_Buffer */
 
 #if !defined(__KERNEL__)
-#include <stdio.h>
+#include "stdio.h"
 #endif
 
 #include "lua.h"
--- a/extensions/LUA/lua/lcode.c
+++ b/extensions/LUA/lua/lcode.c
@@ -4,7 +4,7 @@
 ** See Copyright Notice in lua.h
 */
 
-#include <stdlib.h>
+#include "include/stdlib.h"
 
 #define lcode_c
 #define LUA_CORE
--- a/extensions/LUA/lua/ldebug.c
+++ b/extensions/LUA/lua/ldebug.c
@@ -5,8 +5,8 @@
 */
 
 
-#include <stddef.h>
-#include <string.h>
+#include <linux/stddef.h>
+#include <linux/string.h>
 
 #define ldebug_c
 #define LUA_CORE
--- a/extensions/LUA/lua/ldo.c
+++ b/extensions/LUA/lua/ldo.c
@@ -4,9 +4,9 @@
 ** See Copyright Notice in lua.h
 */
 
-#include <setjmp.h>
-#include <stdlib.h>
-#include <string.h>
+#include "include/setjmp.h"
+#include "include/stdlib.h"
+#include "include/string.h"
 
 #define ldo_c
 #define LUA_CORE
--- a/extensions/LUA/lua/ldump.c
+++ b/extensions/LUA/lua/ldump.c
@@ -4,7 +4,7 @@
 ** See Copyright Notice in lua.h
 */
 
-#include <stddef.h>
+#include <linux/stddef.h>
 
 #define ldump_c
 #define LUA_CORE
--- a/extensions/LUA/lua/lfunc.c
+++ b/extensions/LUA/lua/lfunc.c
@@ -5,7 +5,7 @@
 */
 
 
-#include <stddef.h>
+#include <linux/stddef.h>
 
 #define lfunc_c
 #define LUA_CORE
--- a/extensions/LUA/lua/lgc.c
+++ b/extensions/LUA/lua/lgc.c
@@ -4,7 +4,7 @@
 ** See Copyright Notice in lua.h
 */
 
-#include <string.h>
+#include "include/string.h"
 
 #define lgc_c
 #define LUA_CORE
--- a/extensions/LUA/lua/llex.c
+++ b/extensions/LUA/lua/llex.c
@@ -4,9 +4,9 @@
 ** See Copyright Notice in lua.h
 */
 
-#include <ctype.h>
-#include <locale.h>
-#include <string.h>
+#include "include/ctype.h"
+#include "include/locale.h"
+#include "include/string.h"
 
 #define llex_c
 #define LUA_CORE
--- a/extensions/LUA/lua/llimits.h
+++ b/extensions/LUA/lua/llimits.h
@@ -7,7 +7,7 @@
 #ifndef llimits_h
 #define llimits_h
 
-#include <stddef.h>
+#include <linux/stddef.h>
 
 #include "lua.h"
 
--- a/extensions/LUA/lua/lmem.c
+++ b/extensions/LUA/lua/lmem.c
@@ -5,7 +5,7 @@
 */
 
 
-#include <stddef.h>
+#include <linux/stddef.h>
 
 #define lmem_c
 #define LUA_CORE
--- a/extensions/LUA/lua/lmem.h
+++ b/extensions/LUA/lua/lmem.h
@@ -8,7 +8,7 @@
 #define lmem_h
 
 
-#include <stddef.h>
+#include <linux/stddef.h>
 
 #include "llimits.h"
 #include "lua.h"
--- a/extensions/LUA/lua/lobject.c
+++ b/extensions/LUA/lua/lobject.c
@@ -4,10 +4,10 @@
 ** See Copyright Notice in lua.h
 */
 
-#include <ctype.h>
-#include <stdio.h>
-#include <stdlib.h>
-#include <string.h>
+#include "include/ctype.h"
+#include "include/stdio.h"
+#include "include/stdlib.h"
+#include "include/string.h"
 
 #define lobject_c
 #define LUA_CORE
--- a/extensions/LUA/lua/lobject.h
+++ b/extensions/LUA/lua/lobject.h
@@ -9,7 +9,7 @@
 #define lobject_h
 
 
-#include <stdarg.h>
+#include <linux/stdarg.h>
 
 
 #include "llimits.h"
--- a/extensions/LUA/lua/lparser.c
+++ b/extensions/LUA/lua/lparser.c
@@ -5,7 +5,7 @@
 */
 
 
-#include <string.h>
+#include "include/string.h"
 
 #define lparser_c
 #define LUA_CORE
--- a/extensions/LUA/lua/lstate.c
+++ b/extensions/LUA/lua/lstate.c
@@ -5,7 +5,7 @@
 */
 
 
-#include <stddef.h>
+#include <linux/stddef.h>
 
 #define lstate_c
 #define LUA_CORE
--- a/extensions/LUA/lua/lstring.c
+++ b/extensions/LUA/lua/lstring.c
@@ -4,7 +4,7 @@
 ** See Copyright Notice in lua.h
 */
 
-#include <string.h>
+#include "include/string.h"
 
 #define lstring_c
 #define LUA_CORE
--- a/extensions/LUA/lua/lstrlib.c
+++ b/extensions/LUA/lua/lstrlib.c
@@ -5,11 +5,11 @@
 */
 
 
-#include <ctype.h>
-#include <stddef.h>
-#include <stdio.h>
-#include <stdlib.h>
-#include <string.h>
+#include "include/ctype.h"
+#include "linux/stddef.h"
+#include "include/stdio.h"
+#include "include/stdlib.h"
+#include "include/string.h"
 
 #define lstrlib_c
 #define LUA_LIB
--- a/extensions/LUA/lua/ltable.c
+++ b/extensions/LUA/lua/ltable.c
@@ -18,7 +18,7 @@
 ** Hence even when the load factor reaches 100%, performance remains good.
 */
 
-#include <string.h>
+#include "include/string.h"
 
 #define ltable_c
 #define LUA_CORE
--- a/extensions/LUA/lua/ltablib.c
+++ b/extensions/LUA/lua/ltablib.c
@@ -5,7 +5,7 @@
 */
 
 
-#include <stddef.h>
+#include <linux/stddef.h>
 
 #define ltablib_c
 #define LUA_LIB
--- a/extensions/LUA/lua/ltm.c
+++ b/extensions/LUA/lua/ltm.c
@@ -4,7 +4,7 @@
 ** See Copyright Notice in lua.h
 */
 
-#include <string.h>
+#include "include/string.h"
 
 #define ltm_c
 #define LUA_CORE
--- a/extensions/LUA/lua/lua.h
+++ b/extensions/LUA/lua/lua.h
@@ -9,7 +9,7 @@
 #ifndef lua_h
 #define lua_h
 
-#include <stddef.h>
+#include "linux/stddef.h"
 
 #include "luaconf.h"
 
--- a/extensions/LUA/lua/luaconf.h
+++ b/extensions/LUA/lua/luaconf.h
@@ -8,7 +8,7 @@
 #ifndef lconfig_h
 #define lconfig_h
 
-#include <stddef.h>
+#include <linux/stddef.h>
 
 #if !defined(__KERNEL__)
 #include <limits.h>
@@ -240,7 +240,7 @@
 #define lua_stdin_is_tty()	isatty(0)
 #elif defined(LUA_WIN)
 #include <io.h>
-#include <stdio.h>
+#include "stdio.h"
 #define lua_stdin_is_tty()	_isatty(_fileno(stdin))
 #else
 #define lua_stdin_is_tty()	1  /* assume stdin is a tty */
@@ -282,7 +282,7 @@
 ** GNU readline and history facilities).
 */
 #if defined(LUA_USE_READLINE)
-#include <stdio.h>
+#include "stdio.h"
 #include <readline/readline.h>
 #include <readline/history.h>
 #define lua_readline(L,b,p)	((void)L, ((b)=readline(p)) != NULL)
--- a/extensions/LUA/lua/lundump.c
+++ b/extensions/LUA/lua/lundump.c
@@ -4,7 +4,7 @@
 ** See Copyright Notice in lua.h
 */
 
-#include <string.h>
+#include "include/string.h"
 
 #define lundump_c
 #define LUA_CORE
--- a/extensions/LUA/lua/lvm.c
+++ b/extensions/LUA/lua/lvm.c
@@ -4,9 +4,9 @@
 ** See Copyright Notice in lua.h
 */
 
-#include <stdio.h>
-#include <stdlib.h>
-#include <string.h>
+#include "include/stdio.h"
+#include "include/stdlib.h"
+#include "include/string.h"
 
 #define lvm_c
 #define LUA_CORE
--- a/extensions/LUA/lua/lzio.c
+++ b/extensions/LUA/lua/lzio.c
@@ -4,7 +4,7 @@
 ** See Copyright Notice in lua.h
 */
 
-#include <string.h>
+#include "include/string.h"
 
 #define lzio_c
 #define LUA_CORE
--- a/extensions/LUA/nf_lua.c
+++ b/extensions/LUA/nf_lua.c
@@ -9,9 +9,9 @@
 
 #endif
 
-#include "lua.h"
-#include "lobject.h" /*sizeof(udata) */
-#include "lauxlib.h"
+#include "lua/lua.h"
+#include "lua/lobject.h" /*sizeof(udata) */
+#include "lua/lauxlib.h"
 #include "controller.h"
 
 #if defined(__KERNEL__) /* reachs until luaopen_nflib */
--- a/extensions/pknock/xt_pknock.c
+++ b/extensions/pknock/xt_pknock.c
@@ -28,7 +28,7 @@
 #include <linux/netfilter/x_tables.h>
 #include <crypto/hash.h>
 #include "xt_pknock.h"
-#include "compat_xtables.h"
+#include "../compat_xtables.h"
 
 enum status {
 	ST_INIT = 1,
@@ -296,7 +296,7 @@ static const struct proc_ops pknock_proc
 static void update_rule_gc_timer(struct xt_pknock_rule *rule)
 {
 	if (timer_pending(&rule->timer))
-		del_timer(&rule->timer);
+		timer_delete(&rule->timer);
 	rule->timer.expires = jiffies + msecs_to_jiffies(gc_expir_time);
 	add_timer(&rule->timer);
 }
@@ -355,7 +355,7 @@ has_logged_during_this_minute(const stru
 static void peer_gc(struct timer_list *tl)
 {
 	unsigned int i;
-	struct xt_pknock_rule *rule = from_timer(rule, tl, timer);
+	struct xt_pknock_rule *rule = timer_container_of(rule, tl, timer);
 	struct peer *peer;
 	struct list_head *pos, *n;
 
@@ -517,7 +517,7 @@ remove_rule(struct xt_pknock_mtinfo *inf
 		remove_proc_entry(info->rule_name, pde);
 	pr_debug("(D) rule deleted: %s.\n", rule->rule_name);
 	if (timer_pending(&rule->timer))
-		del_timer(&rule->timer);
+		timer_delete(&rule->timer);
 	list_del(&rule->head);
 	kfree(rule->peer_head);
 	kfree(rule);
